File: boilerplate/java/order-service/src/main/java/com/example/orderservice/application/usecases/ListOrdersUseCaseImpl.java
package com.example.orderservice.application.usecases;

import com.example.orderservice.application.dtos.OrderListItemResult;
import com.example.orderservice.application.dtos.PaginatedResult;
import com.example.orderservice.domain.models.Order;
import com.example.orderservice.domain.models.OrderState;
import com.example.orderservice.domain.ports.OrderRepository;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public class ListOrdersUseCaseImpl implements ListOrdersUseCase {
    private final OrderRepository orderRepository;

    public ListOrdersUseCaseImpl(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    @Override
    public PaginatedResult<OrderListItemResult> execute(UUID customerId, OrderState status, int page, int size) {
        if (page < 0) page = 0;
        if (size < 1) size = 20;
        if (size > 100) size = 100;

        List<Order> orders = orderRepository.findByCustomerId(customerId, status, page, size);
        long total = orderRepository.countByCustomerId(customerId, status);

        List<OrderListItemResult> results = orders.stream()
            .map(o -> new OrderListItemResult(
                o.getId().getValue(),
                o.getCustomerId(),
                o.getStatus(),
                o.calculateTotalValue(),
                o.getCreatedAt(),
                o.getItems().size()
            ))
            .toList();

        return new PaginatedResult<>(results, page, size, total, 0);
    }
}

File: boilerplate/nestjs/order-service/src/application/usecases/list-orders.use-case.impl.ts
import { Injectable, Inject } from '@nestjs/common';

import { IListOrdersUseCase } from './list-orders.use-case.interface';
import { OrderStatus } from '../../domain/models/order-status.enum';
import { OrderRepositoryPort } from '../../domain/ports/order-repository.port';
import { OrderListItemResult, PaginatedResult } from '../dtos/user-order.dto';

@Injectable()
export class ListOrdersUseCaseImpl implements IListOrdersUseCase {
    constructor(
        @Inject('OrderRepositoryPort')
        private readonly orderRepository: OrderRepositoryPort,
    ) {}

    async execute(customerId: string, status: OrderStatus, page: number, size: number): Promise<PaginatedResult<OrderListItemResult>> {
        const p = page < 0 ? 0 : page;
        const s = size < 1 ? 20 : (size > 100 ? 100 : size);

        const orders = await this.orderRepository.findAll({
            skip: p * s,
            take: s,
        });
        const total = await this.orderRepository.countAll();

        const results = orders.map(o => ({
            orderId: o.id.value,
            customerId: o.id.value, // In a real app, customerId would be a separate field
            status: o.status,
            totalAmount: o.totalAmount().toString(),
            createdAt: o.createdAt,
            itemCount: o.items.length,
        }));

        return {
            content: results,
            page: p,
            size: s,
            totalElements: total,
            totalPages: Math.ceil(total / s),
        };
    }
}
